src/Path/PathBuilder.php

```diff
@@ -0,0 +1,142 @@
+<?php /** @noinspection ALL */
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace ArchitectureSniffer\Path;
+
+use ArchitectureSniffer\Path\Transfer\PathTransfer;
+
+class PathBuilder implements PathBuilderInterface
+{
+    protected const CONST_NAME_APPLICATION_ROOT_DIR = 'APPLICATION_ROOT_DIR';
+
+    protected const PATTERN_PATH_MODULE_SCHEMA_FOLDER = 'Persistence/Propel/Schema';
+
+    /**
+     * @param string $filePath
+     *
+     * @return \ArchitectureSniffer\Path\Transfer\PathTransfer
+     */
+    public function getPath(string $filePath): PathTransfer
+    {
+        $rootPath = $this->getRootApplicationDirectoryPathByFilePath($filePath);
+        $corePath = $this->getCorePath($rootPath);
+        $projectPath = $this->getProjectPath($rootPath);
+
+        $pathTransfer = new PathTransfer();
+
+        $pathTransfer->setRootPath($rootPath);
+        $pathTransfer->setCorePath($corePath);
+        $pathTransfer->setProjectPath($projectPath);
+
+        return $pathTransfer;
+    }
+
+    /**
+     * @param string $filePath
+     *
+     * @return string
+     */
+    public function getRootApplicationDirectoryPathByFilePath(string $filePath): string
+    {
+        if ($this->isApplicationRootDefined()) {
+            return ${static::CONST_NAME_APPLICATION_ROOT_DIR};
+        }
+
+        $vendorPosition = strpos($filePath, 'vendor');
+
+        if ($vendorPosition !== false) {
+            return substr($filePath, 0, $vendorPosition);
+        }
+
+        $sourcePosition = strpos($filePath, 'src');
+
+        return substr($filePath, 0, $sourcePosition);
+    }
+
+    /**
+     * @param string $rootPath
+     *
+     * @return string
+     */
+    public function getProjectPath(string $rootPath): string
+    {
+        $path = rtrim($rootPath, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR;
+        $path .= implode(DIRECTORY_SEPARATOR, [
+            'src',
+            'Pyz',
+            'Zed',
+        ]);
+
+        return rtrim($path, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR;
+    }
+
+    /**
+     * @param string $rootPath
+     *
+     * @return string
+     */
+    public function getCorePath(string $rootPath): string
+    {
+        $path = rtrim($rootPath, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR;
+        $path .= implode(DIRECTORY_SEPARATOR, [
+            'vendor',
+            'spryker',
+            'spryker',
+            'Bundles',
```

Comment:
This kind of thing is only relevant and useful for nonsplit way (Spryker internal).
Can we also do this 
a) for standalone modules per namespace (mainly Spryker, SprykerEco, SprykerMiddleware right now)
b) for external split way? So customers can use it.

Reply:
I'll try to fix it right now